neqo-transport/src/send_stream.rs

```diff
@@ -602,20 +571,43 @@ impl SendStream {
     }
 
     pub fn send(&mut self, buf: &[u8]) -> Res<usize> {
-        let sent = match self.state {
-            SendStreamState::Ready => {
-                let mut send_buf = TxBuffer::new();
-                let sent = send_buf.send(buf);
-                self.state.transition(SendStreamState::Send { send_buf });
-                sent
+        if let SendStreamState::Ready = self.state {
+            self.state.transition(SendStreamState::Send {
+                send_buf: TxBuffer::new(),
+            });
+        }
+
+        let stream_credit_avail = self.credit_avail();
+        let conn_credit_avail = self.flow_mgr.borrow().conn_credit_avail();
+        let credit_avail = min(stream_credit_avail, conn_credit_avail);
```

Comment:
I wonder if we should be using `usize` for flow control credit.

Reply:
I don't believe so. It doesn't really matter on 64 bit archs, but on a 32 bit arch that would mean that max credit would be limited to 4GiB, which doesn't seem cool.

Reply:
I think that's entirely cool, but I guess it doesn't matter.

Reply:
probably not but I already changed it :smile: